lang/cs/Org.Apache.REEF.Client/YARN/IYarnREEFClient.cs

```diff
@@ -33,5 +34,13 @@ public interface IYarnREEFClient : IREEFClient
         /// <returns></returns>
         [Unstable("0.17", "Working in progress for rest API id returned")]
         Task<IReadOnlyDictionary<string, IApplicationReport>> GetApplicationReports();
+
+        /// <summary>
+        /// Kills the job application and return Job status
+        /// </summary>
+        /// <param name="appId"></param>
+        /// <returns></returns>
+        [Unstable("0.17", "Working in progress for rest API status returned")]
+        Task<FinalState> KillJobApplication(string appId);
```

Comment:
If AppId is invalid, or if the application has completed,  exception will be thrown from the rest call. We can add code to catch the exception, log it, but I would still like to throw it to the caller for caller to get details of the failure. The reason of the failure is wrapped in the exception details thrown from Yarn.